homeassistant/components/wiz_light/light.py

```diff
@@ -0,0 +1,394 @@
+
+    @property
+    def name(self):
+        """Return the ip as name of the device if any."""
+        return self._name
+
+    @property
+    def unique_id(self):
+        """Return light unique_id."""
+        return self._mac
+
+    @property
+    def is_on(self):
+        """Return true if light is on."""
+        return self._state
+
+    async def async_turn_on(self, **kwargs):
+        """Instruct the light to turn on."""
+        brightness = None
+
+        if ATTR_BRIGHTNESS in kwargs:
+            brightness = kwargs.get(ATTR_BRIGHTNESS)
+
+        if ATTR_RGB_COLOR in kwargs:
+            pilot = PilotBuilder(rgb=kwargs.get(ATTR_RGB_COLOR), brightness=brightness)
+
+        if ATTR_HS_COLOR in kwargs:
+            rgb = color_utils.color_hs_to_RGB(
+                kwargs[ATTR_HS_COLOR][0], kwargs[ATTR_HS_COLOR][1]
+            )
+            pilot = PilotBuilder(rgb=rgb, brightness=brightness)
+        else:
+            colortemp = None
+            if ATTR_COLOR_TEMP in kwargs:
+                kelvin = color_utils.color_temperature_mired_to_kelvin(
+                    kwargs[ATTR_COLOR_TEMP]
+                )
+                colortemp = kelvin
+                _LOGGER.debug(
+                    "[wizlight %s] kelvin changed and send to bulb: %s",
+                    self._light.ip,
+                    colortemp,
+                )
+
+            sceneid = None
+            if ATTR_EFFECT in kwargs:
+                sceneid = self._light.get_id_from_scene_name(kwargs[ATTR_EFFECT])
+
+            if sceneid == 1000:  # rhythm
+                pilot = PilotBuilder()
+            else:
+                pilot = PilotBuilder(
+                    brightness=brightness, colortemp=colortemp, scene=sceneid
+                )
+                _LOGGER.debug(
+                    "[wizlight %s] Pilot will be send with brightness=%s, colortemp=%s, scene=%s",
+                    self._light.ip,
+                    brightness,
+                    colortemp,
+                    sceneid,
+                )
+
+            sceneid = None
+            if ATTR_EFFECT in kwargs:
+                sceneid = self._light.get_id_from_scene_name(kwargs[ATTR_EFFECT])
+
+            if sceneid == 1000:  # rhythm
+                pilot = PilotBuilder()
+            else:
+                pilot = PilotBuilder(
+                    brightness=brightness, colortemp=colortemp, scene=sceneid
+                )
+
+        await self._light.turn_on(pilot)
+
+    async def async_turn_off(self, **kwargs):
+        """Instruct the light to turn off."""
+        await self._light.turn_off()
+
+    @property
+    def color_temp(self):
+        """Return the CT color value in mireds."""
+        return self._temperature
+
+    def get_min_mireds(self) -> int:
+        """Return the coldest color_temp that this light supports."""
+        if self._bulbtype is None:
+            return color_utils.color_temperature_kelvin_to_mired(6500)
+
```

Comment:
It might make things a bit easier and avoid some branching if the lib provided an unknown bulb type with reasonable defaults.    You could warn if the bulb type if the unknown type.